Shuffle around a few things so that Grafx2 compiles on Genode and provide
a dummy implementation for the drive select dialog.
--- src/app/grafx2/src/filesel.c	2012-09-29 01:06:42.000000000 +0200
+++ src/app/grafx2/src/filesel.c	2015-12-29 17:27:50.605573933 +0100
@@ -623,7 +623,9 @@
         drive_index++;
       }
     }
-  
+  #elif defined(GENODE)
+  {
+  }
   #else
   {
     //Sous les différents unix, on va mettre
--- src/app/grafx2/src/main.c	2012-09-30 18:12:26.000000000 +0200
+++ src/app/grafx2/src/main.c	2015-12-29 17:27:50.605573933 +0100
@@ -528,7 +528,7 @@
   
 
   // SDL
-  if(SDL_Init(SDL_INIT_VIDEO|SDL_INIT_JOYSTICK) < 0)
+  if(SDL_Init(SDL_INIT_VIDEO) < 0)
   {
     // The program can't continue without that anyway
     printf("Couldn't initialize SDL.\n");
--- src/app/grafx2/src/mountlist.c	2012-08-11 17:13:36.000000000 +0200
+++ src/app/grafx2/src/mountlist.c	2015-12-29 17:27:50.605573933 +0100
@@ -20,7 +20,7 @@
 */
 
 // This file is not used on some platforms, so don't do anything for them
-#if(!defined(__WIN32__))&&(!defined(__amigaos4__))&&(!defined(__AROS__))&&(!defined(__MORPHOS__))&&(!defined(__amigaos__))
+#if(!defined(__WIN32__))&&(!defined(__amigaos4__))&&(!defined(__AROS__))&&(!defined(__MORPHOS__))&&(!defined(__amigaos__))&&(!defined(GENODE))
 
 // We don't use autoconf and all that in grafx2, so let's do the config here ...
 #if defined(__macosx__) || defined(__FreeBSD__) || defined(__OpenBSD__)                       // MacOS X is POSIX compliant
--- src/app/grafx2/src/realpath.c	2012-08-05 15:18:08.000000000 +0200
+++ src/app/grafx2/src/realpath.c	2015-12-29 17:27:50.601574004 +0100
@@ -6,6 +6,7 @@
 #include <fcntl.h>
 #include <string.h>
 #include <unistd.h>
+#include <limits.h>
 #if defined(__AROS__) || defined(__linux__) || defined(__GLIBC__)
 #include <limits.h>
 #endif
